File: desktop-application/src/main/java/ro/ovidiuconeac/foodtavern/desktopapp/features/food/mvp/presenter/FoodPresenter.java
package ro.ovidiuconeac.foodtavern.desktopapp.features.food.mvp.presenter;

import ro.ovidiuconeac.foodtavern.desktopapp.features.food.rest.RestServiceApi;
import ro.ovidiuconeac.foodtavern.models.features.food.Cheese;
import ro.ovidiuconeac.foodtavern.models.features.food.Fruit;
import ro.ovidiuconeac.foodtavern.models.features.food.Sweet;

/**
 * Created by ovidiu on 2/24/17.
 */
public interface FoodPresenter {
    // Fruits
    void requestFruit(RestServiceApi restServiceApi);
    void requestAllFruits(RestServiceApi restServiceApi);
    void requestAddNewFruit(RestServiceApi restServiceApi, Fruit fruit);
    // Cheeses
    void requestCheese(RestServiceApi restServiceApi);
    void requestAllCheeses(RestServiceApi restServiceApi);
    void requestAddNewCheese(RestServiceApi restServiceApi, Cheese cheese);
    // Sweets
    void requestSweet(RestServiceApi restServiceApi);
    void requestAllSweets(RestServiceApi restServiceApi);
    void requestAddNewSweet(RestServiceApi restServiceApi, Sweet sweet);
}

File: web-application/src/main/resources/public/js/ro/ovidiuconeac/foodtavern/webapp/components/food/mvp/presenter/FoodPresenter.js
/**
 * Created by ovidiu on 3/22/17.
 */
function FoodPresenter(view) {

    var view = view;
    var model = new FoodUseCases(view);

    FoodPresenter.prototype.requestFruit = function() {
        var x = model.getFruit()
        view.postFruit(x);
    }

    FoodPresenter.prototype.requestAllFruits = function() {
        model.getAllFruits();
    }

    FoodPresenter.prototype.requestAddNewFruit = function(fruit) {
        var fruitObject = {};
        fruitObject["name"] = fruit;
        model.addNewFruit(fruitObject);
    }

    FoodPresenter.prototype.requestCheese = function() {
        model.getCheese();
    }

    FoodPresenter.prototype.requestAllCheeses = function() {
        model.getAllCheeses();
    }

    FoodPresenter.prototype.requestAddNewCheese = function(cheese) {
        var cheeseObject = {};
        cheeseObject["name"] = cheese;
        model.addNewCheese(cheeseObject);
    }

    FoodPresenter.prototype.requestSweet = function() {
        model.getSweet();
    }

    FoodPresenter.prototype.requestAllSweets = function() {
        model.getAllSweets();
    }

    FoodPresenter.prototype.requestAddNewSweet = function(sweet) {
        var sweetObjects = {};
        sweetObjects["name"] = sweet;
        model.addNewSweet(sweetObjects);
    }
}